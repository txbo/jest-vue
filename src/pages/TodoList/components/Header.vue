<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        data-test='header-input'
        :value="inputValue"
        @input="e => changeInputValue(e.target.value)"
        @keyup.enter="addTodoItem"
        placeholder="add TodoItem" />
    </div>
  </div>
</template>

<script>
import { mapState, mapMutations } from 'vuex'
export default {
  name: 'Header',
  computed: {
    ...mapState({
      inputValue: state => state.inputValue
    })
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.changeInputValue('')
      }
    },
    ...mapMutations({
      changeInputValue: 'changeInputValue'
    })
  }
}
</script>

<style scoped lang="stylus">
.header
  background #333
  line-height 60px
.header-content
  width 600px
  margin 0 auto
  color #fff
  font-size 24px
.header-input
  float right
  width 360px
  line-height 24px
  margin-top 16px
  outine: none
  color #333
  text-indent 10px
</style>
