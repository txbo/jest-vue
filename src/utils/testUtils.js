export const findTestWrapper = (wrapper, tag) => wrapper.findAll(`[data-test="${tag}"]`)
