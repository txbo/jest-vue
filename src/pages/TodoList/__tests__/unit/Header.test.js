import { shallowMount } from '@vue/test-utils'
import Header from '@/pages/TodoList/components/Header'
import { findTestWrapper } from '@/utils/testUtils'
import store from '@/store'

describe('Header.vue', () => {
  it('样式发生改变', () => {
    const wrapper = shallowMount(Header, { store })
    expect(wrapper).toMatchSnapshot()
  })
  it('包含 input 框', () => {
    const wrapper = shallowMount(Header, { store })
    const input = findTestWrapper(wrapper, 'header-input')
    expect(input.exists()).toBeTruthy()
  })
  it('input 框初始值为空', () => {
    const wrapper = shallowMount(Header, { store })
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe('')
  })

  it('input 框值发生变化，数据应该跟着变', () => {
    const wrapper = shallowMount(Header, { store })
    const input = findTestWrapper(wrapper, 'header-input')
    input.setValue('hello world')
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe('hello world')
  })
  it('input 框键入回车时无内容，无反应', () => {
    const wrapper = shallowMount(Header, { store })
    const input = findTestWrapper(wrapper, 'header-input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })
  it('input 框键入回车时有内容，触发事件，清空 input', () => {
    const wrapper = shallowMount(Header, { store })
    const input = findTestWrapper(wrapper, 'header-input')
    input.setValue('hello world')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.inputValue).toBe('')
  })
})
