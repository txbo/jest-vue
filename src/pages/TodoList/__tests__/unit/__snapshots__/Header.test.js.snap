// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue 样式发生改变 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="header-input" placeholder="add TodoItem" class="header-input"></div>
</div>
`;
